app/models/merchant.rb

```diff
@@ -8,15 +8,17 @@ class Merchant < ApplicationRecord
   has_many :customers, through: :invoices
 
   def top_5_customers
-    customers.joins(:transactions)
-    .where(transactions: { result: "success" })
-    .group(:id)
-    .order("COUNT(transactions.id) DESC")
-    .limit(5)
+    customers
+      .select('customers.*, COUNT(transactions.id) as success_transacts')
```

Comment:
I rewrote this method so that the active record query creates table of customers and a count of their transactions for a given merchant, then it orders that list and limits to top 5.